### PR TITLE
Switch build to x-caddy

### PR DESCRIPTION
**Pull request by lucaslorentz**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +0,0 @@
-module github.com/lucaslorentz/caddy-docker-proxy/v2
```
Comment (francislavoie) [inline go.mod:1]: Don't you still need the `go.mod` so others can install this as a plugin?

Comment (francislavoie) [reply go.mod:1]: Oh nevermind, scratch that. I see https://github.com/lucaslorentz/caddy-docker-proxy/blob/master/plugin/go.mod :sweat_smile:

Comment (lucaslorentz) [reply go.mod:1]: There is a go.mod inside plugin folder.

**Comment by lucaslorentz** [comment]
Fixed https://github.com/lucaslorentz/caddy-docker-proxy/issues/141

**Pull request merged**